### ISSUE TITLE
doc switcher right click column header (ex:name) causes crash

### ISSUE THREAD

**Issue by theotang** [notepad-plus-plus/notepad-plus-plus#5479]
### Description of the Issue

Right click column header in Doc Switcher causes crash

### Steps to Reproduce the Issue

1. Enable Doc Switcher
2. Right click column header, ex: Name
3. Crash

### Expected Behavior

Not crash

### Actual Behavior

Crash

**Comment by MetaChuh** [comment]
welcome to the notepad++ issue tracker, @theotang 

**i can not reproduce your experience on notepad++ 7.6.5. 32 and 64 bit.**

right clicking on any doc switcher column header, or anywhere else inside the doc switcher, does not produce a crash.

unfortunately we do not have your required debug information to verify your described behaviour under similar conditions.

**please test the following, to verify if a clean notepad++ (without any plugins that could cause this) would work on your system.**

* download the notepad++ 7.6.5. portable version from [>>> here <<<](https://notepad-plus-plus.org/repository/7.x/7.6.5/npp.7.6.5.bin.zip).
(the portable version does not require any installation and runs completely independent and isolated from your installed version. all custom settings, as well as any stuff you might want to try at the portable version, will not modify your installed version)

* extract `npp.7.6.5.bin.zip` to e.g. your desktop.

* open the extracted `npp.7.6.5.bin` folder and start this portable version by double-clicking on `notepad++.exe` inside this folder.
(note: notepad++.exe will just be seen as notepad++, if you have enabled to hide all known file extensions at your explorer settings)

-----

```
general notes:

please close your issue as soon as you are satisfied,
to help keeping the notepad++ issue tracker clean.

when creating a new issue, please provide all information as shown at the issue template.
your notepad++ debug info is part of the minimum requirements.
(? > Debug Info... > Copy debug info into clipboard).

for general questions, or if it is not sure, whether your issue is directly related to
the notepad++ source code, please visit us at the notepad++ community forum, and first
search if a related topics exists. you are welcome to post at either an eventually similar
topic, or create a new topic at Help Wanted or General Discussion of your choice.

```
[<kbd> &emsp; &emsp; &emsp; &emsp; &emsp; &emsp; &emsp; &emsp; &emsp; &emsp; &emsp; &emsp; Click here to visit the Notepad++ Community Forum &emsp; &emsp; &emsp; &emsp; &emsp; &emsp;&emsp; &emsp; &emsp; &emsp; &emsp; &emsp; &emsp; </kbd>](https://notepad-plus-plus.org/community/)

**Comment by sasumner** [comment]
@MetaChuh 

>the portable version does not require any installation and runs completely independent and isolated from your installed version

I'm wondering, if, when you give this advice, that you should point out that you have to end your normal N++ before firing up a portable version.  Those that are noobish might not know this, and would simply be switched to their already running instance when they think they are running portable.  Just a thought.

**Comment by MetaChuh** [comment]
@sasumner 

yes, you are correct as usual.

**please test the following, to verify, if a clean notepad++ (without any plugins that could cause this) would work on your system.**

* download the notepad++ 7.6.5. portable version from [>>> here <<<](https://notepad-plus-plus.org/repository/7.x/7.6.5/npp.7.6.5.bin.zip).
(the portable version does not require any installation and runs completely independent and isolated from your installed version. all custom settings, as well as any stuff you might want to try at the portable version, will not modify your installed version)

* extract `npp.7.6.5.bin.zip` to e.g. your desktop.

* **important note:** make sure to close all instances of notepad++, before launching the portable notepad++ version at the next step.

* open the extracted `npp.7.6.5.bin` folder and start this portable version by double-clicking on `notepad++.exe` inside this folder.
(note: notepad++.exe will just be seen as notepad++, if you have enabled to hide all known file extensions at your explorer settings)

-----

ps: is this revised version better, or would you change the order or any text ?
maybe i could need your help to simplify the wording, as i'm so used to that guide, that i already might be "verbally colourblind". 😉

**Comment by theotang** [comment]
odd, still crashes for me.

debug info

Notepad++ v7.6.5   (64-bit)
Build time : Mar 29 2019 - 02:43:47
Path : C:\SyncThing\Sync\Portable\Notepad++\notepad++.exe
Admin mode : OFF
Local Conf mode : ON
OS : Windows 10 (64-bit)
Plugins : ComparePlugin.dll NppFavorites.dll PluginManager.dll SessionMgr.dll

**Comment by sasumner** [comment]
@theotang How did those plugins get there??

**Comment by theotang** [comment]
i added those plugins.

i did more testing as advised.  a duplicate npp portable does not crash with same plugins.  here is the duplicate debug info

Notepad++ v7.6.5   (64-bit)
Build time : Mar 29 2019 - 02:43:47
Path : C:\Users\theotang\Downloads\npp.7.6.5.bin.x64\notepad++.exe
Admin mode : OFF
Local Conf mode : ON
OS : Windows 10 (64-bit)
Plugins : ComparePlugin.dll NppFavorites.dll PluginManager.dll SessionMgr.dll 

it appears to crash when my session is loaded.  my session contains 47 files.  do you need me to post my session save file?

**Comment by theotang** [comment]
actually 45 items

**Comment by sasumner** [comment]
7.6.5 has a (recently) known session issue and has been retracted.  Was your original complaint against 7.6.5 (you didn't specify)?

**Comment by theotang** [comment]
yes, thanks.  it is for 7.6.5.  it is more of an annoyance but something i can avoid doing.

**Comment by MetaChuh** [comment]
fyi:

here is the report about the session bug in 7.6.5, including steps to reproduce it every time:
https://notepad-plus-plus.org/community/topic/17395/rolling-back-to-v7-6-4

7.6.6. is on the way

ps: my apologies for not relating doc switcher directly to the session issue.

**Comment by rddim** [comment]
@theotang try with this build [1.0.2800](https://ci.appveyor.com/project/donho/notepad-plus-plus/builds/23550782/job/q8lw9k898k1bbtx0/artifacts) from appveyor. Just download only the `.exe` file and place it in your current portable or regular folder of N++.

**Comment by theotang** [comment]
1.0.2800 still crashes when i right click.  it was placed in a duplicate npp portable folder.  i did not rename the file but launched it.  debug info:

Notepad++ v7.6.5   (64-bit)
Build time : Apr  2 2019 - 21:33:02
Path : C:\Users\theotang\Downloads\npp.7.6.5.bin.x64\Notepad++.X64.Unicode+Release.exe
Admin mode : OFF
Local Conf mode : ON
OS : Windows 10 (64-bit)
Plugins : ComparePlugin.dll NppFavorites.dll PluginManager.dll SessionMgr.dll 

same plugins.  loaded my saved sessions (45 items), right click, crash

**Comment by sasumner** [comment]
You've got to get rid of the plugins for this test.  Especially PluginManager, which is a totally invalid plugin for Notepad++ 7.6+, but really all plugins that don't ship with N++ have to be out for this test.

**Comment by theotang** [comment]
sorry, did not know that.  tried again without any plugins, still crashed

Notepad++ v7.6.5   (64-bit)
Build time : Apr  2 2019 - 21:33:02
Path : C:\Users\theotang\Downloads\npp.7.6.5.bin.x64\Notepad++.X64.Unicode+Release.exe
Admin mode : OFF
Local Conf mode : ON
OS : Windows 10 (64-bit)
Plugins : none

**Comment by MetaChuh** [comment]
@theotang 

> it appears to crash when my session is loaded. my session contains 47 files. do you need me to post my session save file?

until now we did not reply to your offer, as we would have to recreate your folder structures to simulate if it is something within that session file, but i agree with you, that it would be a good idea to have your sessions file that is loaded, while you experience the crash with a right click on the doc switcher column header.

also a screenshot of the crash would be helpful, if you see any exception error message.

note:
we have not been able to reproduce any session crashes with 1.0.2800,  at least not yet.

**Comment by theotang** [comment]
Notepad++ v7.6.5   (64-bit)
Build time : Apr  2 2019 - 21:33:02
Path : C:\Users\theotang\Downloads\npp.7.6.5.bin.x64\Notepad++.X64.Unicode+Release.exe
Admin mode : OFF
Local Conf mode : ON
OS : Windows 10 (64-bit)
Plugins : none

here's how to recreate it, add 34 new blank docs, then right click the header.  it crashes without any message.

**Comment by rddim** [comment]
Confirm! In my case crash with only one blank or not blank doc with or without plugins. Didn't test with 32-bit version, because it is removed from the web.

### Debug Information ###
Notepad++ v7.6.5   (64-bit)
Build time : Apr  2 2019 - 21:33:02
Path : D:\Install\Office Programs\npp.7.6.5.bin.x64\Notepad++.X64.Unicode+Release.exe
Admin mode : ON
Local Conf mode : ON
OS : Windows 7 (64-bit)
Plugins : none

**Comment by MetaChuh** [comment]
@rddim 

strange, i still can not reproduce any crash when right clicking at the name header, ext header, title, nor anywhere at the dock switcher. (please see video below)

what am i missing ?
can you give me a step by step reproduction guide ?

@theotang 
one question for better understanding: right clicking the dock switcher header or title bar does not have any function, why do you right click on it ?

<kbd>debug info:</kbd>
Notepad++ v7.6.5   (64-bit)
Build time : Apr  2 2019 - 21:33:02
Path : C:\Users\User\Desktop\npp.7.6.5.bin.x64\Notepad++.X64.Unicode+Release.exe
Admin mode : OFF
Local Conf mode : ON
OS : Windows 10 (64-bit)
Plugins : mimeTools.dll NppConverter.dll 

![Imgur](https://i.imgur.com/f8AWzsP.gif)

**Comment by rddim** [comment]
@MetaChuh 

Try to not change the focus when you start N++. When record the video you (and I) first start N++, then start recording and that change the focus. For that when I record I click on a main doc window and after that right click on `Name`
1. Start N++
2. Right click on `Name`

![npp_doc_switcher_crash](https://user-images.githubusercontent.com/8998791/55515902-7293e180-5674-11e9-8f28-053a7293093c.gif)

On a 2nd video (with changed focus) I right/left click on `Name` multiple times and I it works as expected. But when I click on main window and back again it crash. Also when I click on `Ext.` instead in main window it crash, but this is not recorded.

![npp_doc_switcher_crash2](https://user-images.githubusercontent.com/8998791/55515919-7de70d00-5674-11e9-92fc-4d22c872a0a7.gif)

**Comment by MetaChuh** [comment]
thanks @rddim 

very strange: **i can reproduce that on one of my win7 x64 machines**, but the same does not happen on the win 10 x64 machine i used for recording ... both using the same portable 7.6.5. x64 folder !?!?!

**all 32 bit versions are working correctly** and do not try to show a context menu when right clicking on the doc switcher menu bar, they only show a menu at the respective file names, as it wouldn't make sense to have a "close all but this" when right clicking on the name or ext bar. 
(i've sent you the removed 32 bit binary, if you need them for anything)

apparently only the x64 versions 7.6.4 and 7.6.5 are affected.
i've not been able to reproduce that on 7.6.3 x64 and below.

ps: i hate things that are not reproducible on all machines ... now i want it to crash on windows 10 too, but it refuses 😉

**Comment by MetaChuh** [comment]
update:

same issue on 7.6.6:
32 bit is ok, but the 64 bit version produces a crash when right-clicking on the doc switcher name bar.
(see screencast)

![Imgur](https://i.imgur.com/alAIITe.gif)

Notepad++ v7.6.6   (64-bit)
Build time : Apr  3 2019 - 23:52:32
Path : C:\Users\User\Desktop\npp.7.6.6.bin.x64\notepad++.exe
Admin mode : OFF
Local Conf mode : ON
OS : Windows 7 (64-bit)
Plugins : mimeTools.dll NppConverter.dll 

ps: i'm still not able to reproduce this crash on my windows 10 machine😤

**Comment by theotang** [comment]
sorry for all the trouble.  i thought i could sort the file list by right click the header.  i was just playing around after i found this feature.  originally, i just wanted side tabs that were horizontal.

i thought i would report it because otherwise, npp has been very stable for me for years.

**Comment by rddim** [comment]
@MetaChuh thanks for the files
@theotang don't worry, it is good to now about this bug because some incidental click may destroy all your works

New tests
* At home PC with clean portable versions
  * v7.6.5 (32 bit) doesn't crash, not show a context menu
  * v7.6.5 (64 bit) crash, show a context menu
  * v7.6.6 (32 bit) doesn't crash, not show a context menu
  * v7.6.6 (64 bit) crash, show a context menu
* At office laptop with clean portable versions
  * v7.6.5 (32 bit) doesn't crash, not show a context menu
  * v7.6.5 (64 bit) doesn't crash, show a context menu
  * v7.6.6 (32 bit) doesn't crash, not show a context menu
  * v7.6.6 (64 bit) doesn't crash, show a context menu

**Comment by donho** [comment]
@theotang  @rddim @MetaChuh 

Could you check this 64 release binary : 
https://notepad-plus-plus.org/temp/notepadpp64.zip

It doesn't contain any fix, but only the right click is disabled for localizing the problem.
THX

**Comment by donho** [comment]
OK, fixed I guess (with the non-fixed binaries, I can reproduce it only randomly).
https://notepad-plus-plus.org/temp/nppFixDocSwitcherCrash.zip

Please confirm me if bug is fixed.

**Comment by MetaChuh** [comment]
fixed on my test machine.

<kbd>Notepad++ v7.6.5   (64-bit)
Build time : Apr  4 2019 - 13:22:50
Path : C:\Users\User\Desktop\npp.7.6.6.bin.x64 fix test\notepad++ donho nppFixDocSwitcherCrash.exe
Admin mode : OFF
Local Conf mode : ON
OS : Windows 7 (64-bit)
Plugins : mimeTools.dll NppConverter.dll </kbd>

@rddim many thanks again, your home pc might give better confirmation, as your results were non random. 👍🙏

@theotang 
your w10 live machine test results will be the most important and appreciated.
**note**: right click is now disabled on doc-switcher name/ext bar only.
ps: thanks for your detailed information and patience.

**Comment by theotang** [comment]
does not crash anymore for me.

Notepad++ v7.6.5   (64-bit)
Build time : Apr  4 2019 - 13:22:50
Path : C:\Users\theotang\Downloads\npp.7.6.5.bin.x64\notepad++.exe
Admin mode : OFF
Local Conf mode : ON
OS : Windows 10 (64-bit)
Plugins : none

**Comment by rddim** [comment]
I'm at home now and the build with disabled right click doesn't crash, at office too. Both machines are running Win 7 x64

Notepad++ v7.6.5   (64-bit)
Build time : Apr  4 2019 - 13:22:50
Path : D:\Install\Office Programs\npp.7.6.6.bin.x64\notepad++.exe
Admin mode : ON
Local Conf mode : ON
OS : Windows 7 (64-bit)
Plugins : mimeTools.dll NppConverter.dll

**Comment by theotang** [comment]
this version crashes

Notepad++ v7.6.6   (64-bit)
Build time : Apr  3 2019 - 23:52:32
Path : C:\SyncThing\Sync\Portable\Notepad++\notepad++.exe
Admin mode : ON
Local Conf mode : ON
OS : Windows 7 (64-bit)
Plugins : ComparePlugin.dll mimeTools.dll NppConverter.dll NppFavorites.dll PluginManager.dll SessionMgr.dll

**Comment by sasumner** [comment]
@theotang 

Yes, at this point that is to be expected.  A change has been committed that will fix it, thus the issue has been closed, but a release hasn't been made with the fix in it yet.  Make sense?

**Comment by theotang** [comment]
also crashes

Notepad++ v7.6.6   (64-bit)
Build time : Apr  3 2019 - 23:52:32
Path : C:\Users\theotang\Downloads\npp.7.6.5.bin.x64\notepad++.exe
Admin mode : OFF
Local Conf mode : ON
OS : Windows 10 (64-bit)
Plugins : none

**Comment by theotang** [comment]
sorry, ok, never mind then!  thanks for the help

**Comment by rddim** [comment]
@theotang use this build [1.0.2809](https://ci.appveyor.com/project/donho/notepad-plus-plus/builds/23606042/job/e8tsm9wolab9jr0s/artifacts) or wait for the next official release

**Comment by MetaChuh** [comment]
hi @theotang 

our apologies for the confusion.

issues will be closed as soon as they are fixed at the source code, not at release date.
(this helps not forgetting to close an abandoned, open issue due to a longer "incubation" time in between.)

please use the build provided by @rddim until the next version, either 7.6.7. (boeing 😉) or minor release is out, and feel absolutely free to ask anything about this issue, even if it is closed for the project itself.

debug for 1.0.2809
<kbd>Notepad++ v7.6.6   (64-bit)
Build time : Apr  4 2019 - 18:02:05
Path : C:\Users\User\Desktop\npp.7.6.6.bin.x64 fix test\Notepad++.X64.7.6.6b1.0.2809 (#5479 donho, Fix crash while right clicking on DocSwitcher's column bar).exe
Admin mode : OFF
Local Conf mode : ON
OS : Windows 7 (64-bit)
Plugins : mimeTools.dll NppConverter.dll</kbd>

best regards and many thanks for finding this bug.

**Comment by sasumner** [comment]
> use this build 1.0.2809 or wait for the next official release

Really?  How about just not doing the action that causes the crash until a real release happens?  Or EVER, because the action doesn't do anything anyway (even in a non-crashing version). :)

@MetaChuh You might be taking this "support" thing a bit too far. :)  Nono, don't take offense, it's great for the N++ user community.

**Comment by rddim** [comment]
Yes, really, why not? What is the problem with this build?

**Comment by sasumner** [comment]
@rddim Ok, ok...

**Comment by dail8859** [comment]
@rddim These are ok to make sure the bug is fixed but I would not suggest anyone use them. These automated builds don't use boost for the regular expression engine, which could cause other issues like affecting the function list parsing, url highlighting, and of course using regex for find/replace.
